syntax = "proto3";

package fricon;

service DataStorage {
  rpc Create (CreateRequest) returns (CreateResponse) {}
  rpc Write (stream WriteRequest) returns (WriteResponse) {}
  rpc UpdateMetadata (UpdateMetadataRequest) returns (UpdateMetadataResponse) {}
}

message CreateRequest {
  Metadata metadata = 1;
}

message CreateResponse {
  int64 id = 1;
  Uuid write_token = 2;
}

message UpdateMetadataRequest {
  int64 id = 1;
  Metadata metadata = 2;
}

message UpdateMetadataResponse {}

message Metadata {
  optional string name = 1;
  optional string description = 2;
  repeated string tags = 3;
}

message Uuid {
  fixed64 msb = 1;
  fixed64 lsb = 2;
}

message WriteRequest {
  bytes record_batch = 1;
}

message WriteResponse {}
